### PR TITLE
feat: support both command array and command string + args formats

### LINKED ISSUE #1: Support both `command: []` array and `command: string` + `args: []` formats for MCP definitions
### Problem

The skill-embedded MCP parser currently expects a specific format for MCP command definitions, but there are two common formats in the ecosystem that should both be supported:

1. **OpenCode format** - Uses a single `command` array containing the executable and all arguments
2. **oh-my-opencode format** - Uses separate `command` (string) and `args` (array) fields

This incompatibility means skills written for one ecosystem don't work in the other without modification.

### Current Behavior

When using a skill definition like:

```yaml
mcp:
  playwright:
    command: ["npx", "-y", "@playwright/mcp@latest"]
```

The parser fails with:
```
TypeError: null is not an object (evaluating 'config.command')
```

### Expected Behavior

Both formats should work interchangeably:

**Format A (OpenCode-style combined array):**
```yaml
mcp:
  playwright:
    command: ["npx", "-y", "@playwright/mcp@latest"]
```

**Format B (oh-my-opencode-style separate fields):**
```yaml
mcp:
  playwright:
    command: npx
    args: ["-y", "@playwright/mcp@latest"]
```

#### OpenCode MCP Server Configuration
From [opencode.ai/docs/mcp-servers](https://opencode.ai/docs/mcp-servers):

```json
{
  "mcp": {
    "my-local-mcp-server": {
      "type": "local",
      "command": ["npx", "-y", "my-mcp-command"],
      "enabled": true
    }
  }
}
```

> The command is how the local MCP server is started. You can also pass in a list of environment variables as well.

#### oh-my-opencode Skill-Embedded MCP
From [oh-my-opencode README](https://github.com/code-yeongyu/oh-my-opencode#skill-embedded-mcp-support):

```yaml
---
description: Browser automation skill
mcp:
  playwright:
    command: npx
    args: ["-y", "@anthropic-ai/mcp-playwright"]
---
```

### Proposed Solution

Normalize both formats internally during parsing:

\`\`\`typescript
function normalizeCommand(config: McpConfig): { command: string; args: string[] } {
  if (Array.isArray(config.command)) {
    // OpenCode format: command: ["npx", "-y", "package"]
    const [cmd, ...args] = config.command;
    return { command: cmd, args };
  } else if (typeof config.command === 'string') {
    // oh-my-opencode format: command: "npx", args: ["-y", "package"]
    return { command: config.command, args: config.args || [] };
  }
  throw new Error('Invalid MCP command configuration');
}
\`\`\`

### Benefits

1. **Cross-ecosystem compatibility** - Skills work in both OpenCode and oh-my-opencode without modification
2. **User flexibility** - Authors can use whichever format they prefer
3. **Reduced friction** - No need to convert skill definitions when switching tools

### PR DESCRIPTION
**Pull request by keybrdist**
## Summary

- Adds `normalizeCommand()` utility to unify both MCP command formats
- Updated `McpServerConfig` type to include optional `args` field
- Added vitest test suite with 9 tests covering both formats and edge cases

## Formats Supported

**OpenCode format (command as array):**
```yaml
mcp:
  playwright:
    command: ["npx", "-y", "@playwright/mcp@latest"]
```

**oh-my-opencode format (command string + args):**
```yaml
mcp:
  playwright:
    command: npx
    args: ["-y", "@anthropic-ai/mcp-playwright"]
```

## Testing

```bash
npm test
# 9 tests pass
```

Closes #1

### CODE REVIEW

**Comment by keybrdist** [comment]
@keybrdist are environments handled as well? I believe there was a slight variation between omo and opencode re: env vs. environment. Maybe I'm wrong so please check.

**Comment by keybrdist** [comment]
Good catch! You're right — oh-my-opencode uses `env` while OpenCode uses `environment`.

Current state:
- `command` + `args` → handled ✅
- `env` vs `environment` → **not handled** ❌

I'll add normalization for this as well. The fix will:
1. Accept both `env` and `environment` in `McpServerConfig`
2. Normalize to `environment` internally (matching OpenCode's convention)
3. Add tests for the env normalization

File: src/__tests__/normalize-command.test.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        command: ['npx', '-y', '@some/package'],
+        args: ['should', 'be', 'ignored']
+      }
+
+      const result = normalizeCommand(config)
+
+      expect(result.command).toBe('npx')
+      expect(result.args).toEqual(['-y', '@some/package'])
+    })
+  })
```
[the remark this answers is not in the document]
Comment (keybrdist) [reply src/__tests__/normalize-command.test.ts]: Fixed in 042955f - added test for empty command array

File: src/utils/env-vars.ts [hunk 2]
```diff
@@ -75,3 +75,49 @@ export function createCleanMcpEnvironment(
...
+
+  if (Array.isArray(envValue)) {
+    const env: Record<string, string> = {}
+    for (const entry of envValue) {
+      const eqIndex = entry.indexOf('=')
+      if (eqIndex > 0) {
+        const key = entry.slice(0, eqIndex)
+        const value = entry.slice(eqIndex + 1)
+        env[key] = value
+      }
```
[the remark this answers is not in the document]
Comment (keybrdist) [reply src/utils/env-vars.ts]: Silently skipping is intentional - consistent with how we handle other malformed entries (no equals sign). Added test in 9b2f874 to document this behavior.
